syntax = "proto3";

package endpoints.word_service;

service WordService {
  rpc DefineWord(DefineWordRequest) returns (DefineWordResponse) {}
}

message WordDefinition {
  string word = 1;
  string definition = 2; 
  string pos = 3;
  repeated string examples = 4;
  repeated string syllables = 5;
}

message DefineWordRequest {
  string word = 1;
}

message DefineWordResponse {
  WordDefinition word = 1;
  bool is_fallback = 2;
}